cpp/bench/ann/src/common/benchmark.hpp

```diff
@@ -287,15 +287,14 @@ void bench_search(::benchmark::State& state,
     std::make_shared<buf<std::size_t>>(current_algo_props->query_memory_type, k * query_set_size);
 
   cuda_timer gpu_timer;
-  auto start = std::chrono::high_resolution_clock::now();
   {
     nvtx_case nvtx{state.name()};
+    [[maybe_unused]] auto ntx_lap = nvtx.lap();
+    [[maybe_unused]] auto gpu_lap = gpu_timer.lap();
+    auto start                    = std::chrono::high_resolution_clock::now();
```

Comment:
I guess you intended to move the initialization of the `start` past the copy of the `algo`?